pkg/sbom/io/encode.go

```diff
@@ -173,8 +173,15 @@ func (e *Encoder) encodePackages(parent *core.Component, result types.Result) {
 	// Group vulnerabilities by package ID
 	vulns := make(map[string][]core.Vulnerability)
 	for _, vuln := range result.Vulnerabilities {
-		v := e.vulnerability(vuln)
-		vulns[v.PkgID] = append(vulns[v.PkgID], v)
+		// There are cases when `Result` contains the same vulnerabilities for the same packages but with different file paths
+		// We don't need to include duplicates.
+		vv := vulns[vuln.PkgID]
+		if _, ok := lo.Find(vv, func(v core.Vulnerability) bool {
```

Comment:
My two cents is that the use of sets is appropriate here. Or duplicates can be removed later using `lo.MapValues` and `lo.UniqBy`

Reply:
hmm... you're right.
There is no point in checking the slice after adding each element.
It is better to check each slice once after adding all vulnerabilities.
Thanks!

updated in https://github.com/aquasecurity/trivy/pull/7023/commits/5aa8ccc9d1aac7f432303ae620c77319fff2513f